sklearn/neighbors/_base.py

```diff
@@ -38,39 +38,39 @@
 from ..utils.validation import check_non_negative
 from ..utils._param_validation import Interval, StrOptions
 from ..utils.parallel import delayed, Parallel
-from ..utils.fixes import parse_version, sp_version
+from ..utils.fixes import parse_version, sp_base_version, sp_version
 from ..exceptions import DataConversionWarning, EfficiencyWarning
 
+SCIPY_METRICS = [
+    "braycurtis",
+    "canberra",
+    "chebyshev",
+    "correlation",
+    "cosine",
+    "dice",
+    "hamming",
+    "jaccard",
+    "mahalanobis",
+    "matching",
+    "minkowski",
+    "rogerstanimoto",
+    "russellrao",
+    "seuclidean",
+    "sokalmichener",
+    "sokalsneath",
+    "sqeuclidean",
+    "yule",
+]
+if sp_base_version < parse_version("1.11"):
+    # Deprecated in SciPy 1.9 and removed in SciPy 1.11
+    SCIPY_METRICS += ["kulsinski"]
```

Comment:
This does not preserves the lexical order.

I think we can insert the string in the list as indicated above or resort the list.